server/notifications.model.js

```diff
@@ -0,0 +1,66 @@
+const mongoose = require('mongoose');
+const _ = require('lodash');
+
+const NotificationSchema = new mongoose.Schema({
+	dismissKey : { type: String, unique: true, required: true },
+	title      : { type: String, default: '' },
+	text       : { type: String, default: '' },
+	createdAt  : { type: Date, default: Date.now },
+	startAt    : { type: Date, default: Date.now },
+	stopAt     : { type: Date, default: Date.now },
+}, { versionKey: false });
+
+NotificationSchema.statics.addNotification = async function(data) {
+	if(!data.dismissKey) return { success: false, message: 'Dismiss key is required!' };
+
+	const defaults = {
+		title   : '',
+		text    : '',
+		startAt : new Date(),
+		stopAt  : new Date(),
+	};
+
+	const notificationData = _.defaults(data, defaults);
+
+	if(!(notificationData.startAt instanceof Date) || !(notificationData.stopAt instanceof Date)) {
```

Comment:
Is it even possible for the wrong date type to be sent? I'm trying to figure out how to manipulate the input box to get this error case but struggling.

Reply:
Good point, besides this is not open to the public, so no one should be exploiting anything or wandering without knowledge.